lib/active_encode/engine_adapters/ffmpeg_adapter.rb

```diff
@@ -51,18 +52,35 @@ def create(input_url, options = {})
 
         new_encode.input = build_input new_encode
 
-        if new_encode.input.duration.blank?
-          new_encode.state = :failed
-          new_encode.percent_complete = 1
+        # Log error if file is empty or inaccessible
+        if new_encode.input.duration.blank? && new_encode.input.file_size.blank?
+          file_error(new_encode, input_url)
+          return new_encode
+        # If file is not empty, try copying file to generate missing metadata
+        elsif new_encode.input.duration.blank? && new_encode.input.file_size.present?
+          filepath = clean_url.to_s.gsub(/\?.*/, '')
+          copy_url = filepath.gsub(filepath, "#{File.basename(filepath, File.extname(filepath))}_temp#{File.extname(filepath)}")
+          copy_path = working_path(copy_url, new_encode.id)
+
+          # -map 0 sets ffmpeg to copy all available streams.
+          # -c copy sets ffmpeg to copy all codecs
+          # -y automatically overwrites the temp file if one already exists
+          `#{FFMPEG_PATH} -loglevel level+fatal -i \"#{input_url}\" -map 0 -c copy -y \"#{copy_path}\"`
+
+          # If ffmpeg copy fails, log error because file is either not a media file
+          # or the file extension does not match the codecs used to encode the file
+          unless $CHILD_STATUS.success?
+            file_error(new_encode, input_url)
+            return new_encode
+          end
 
-          new_encode.errors = if new_encode.input.file_size.blank?
-                                ["#{input_url} does not exist or is not accessible"]
-                              else
-                                ["Error inspecting input: #{input_url}"]
-                              end
+          # Write the mediainfo output to a temp file to preserve metadata from original file
+          `#{MEDIAINFO_PATH} #{curl_option} --Output=XML --LogFile=#{working_path("temp_input_metadata", new_encode.id)} "#{copy_path}"`
```

Comment:
I think we should come up with a better name than "temp_input_metadata" but I'm having problems thinking of a good one.  The file isn't temporary since we need it for the fixed duration.  How about "fixed_duration_input_metadata" or "duration_input_metadata"?